babel/units.py

```diff
@@ -0,0 +1,241 @@
+    >>> format_unit(1200, 'pressure-inch-hg', locale='nb')
+    u'1\\xa0200 tommer kvikks\\xf8lv'
+
+    Number formats may be overridden with the ``format`` parameter.
+
+    >>> from babel._compat import Decimal
+    >>> format_unit(Decimal("-42.774"), 'temperature-celsius', 'short', format='#.0', locale='fr')
+    u'-42,8 \\xb0C'
+
+    The locale's usual pluralization rules are respected.
+
+    >>> format_unit(1, 'length-meter', locale='ro_RO')
+    u'1 metru'
+    >>> format_unit(0, 'length-picometer', locale='cy')
+    u'0 picometr'
+    >>> format_unit(2, 'length-picometer', locale='cy')
+    u'2 bicometr'
+    >>> format_unit(3, 'length-picometer', locale='cy')
+    u'3 phicometr'
+
+    >>> format_unit(15, 'length-horse', locale='fi')
+    Traceback (most recent call last):
+        ...
+    UnknownUnitError: length-horse is not a known unit in fi
+
+    .. versionadded:: 2.2.0
+
+    :param value: the value to format.
+    :param measurement_unit: the code of a measurement unit.
+                             Known units can be found in the CLDR Unit Validity XML file:
+                             http://unicode.org/repos/cldr/tags/latest/common/validity/unit.xml
+    :param length: "short", "long" or "narrow"
+    :param format: An optional format, as accepted by `format_decimal`.
+    :param locale: the `Locale` object or locale identifier
+    """
+    locale = Locale.parse(locale)
+
+    q_unit = _find_unit_pattern(measurement_unit, locale=locale)
+    if not q_unit:
+        raise UnknownUnitError(unit=measurement_unit, locale=locale)
+    unit_patterns = locale._data["unit_patterns"][q_unit].get(length, {})
+    formatted_value = format_decimal(value, format, locale)
+    plural_form = locale.plural_form(value)
+    if plural_form in unit_patterns:
+        return unit_patterns[plural_form].format(formatted_value)
+
+    # Fall back to a somewhat bad representation.
+    return '%s %s' % (
+        formatted_value,
+        (get_unit_name(measurement_unit, length=length, locale=locale) or measurement_unit)
+    )
+
+
+def _find_compound_unit(numerator_unit, denominator_unit, locale=LC_NUMERIC):
+    """
+    Find a predefined compound unit pattern.
+
+    Used internally by format_compound_unit.
+
+    >>> _find_compound_unit("kilometer", "hour", locale="en")
+    'speed-kilometer-per-hour'
+
+    >>> _find_compound_unit("mile", "gallon", locale="en")
+    'consumption-mile-per-gallon'
+
+    :param numerator_unit: The numerator unit's identifier
+    :param denominator_unit: The denominator unit's identifier
+    :param locale: the `Locale` object or locale identifier
+    :return: A key to the `unit_patterns` mapping, or None.
+    """
+    locale = Locale.parse(locale)
+    numerator_unit_length = _find_unit_pattern(numerator_unit, locale=locale)
+    denominator_unit_length = _find_unit_pattern(denominator_unit, locale=locale)
+    if not (numerator_unit_length and denominator_unit_length):
+        return None
```

Comment:
Nit: Maybe mention that the numerator and denominator values we'd have in this case are "length-kilometer" and "duration-hour"